react/src/Searchbar/Searchbar.tsx

```diff
@@ -103,14 +104,19 @@ export const Searchbar = forwardRef<SearchbarProps, 'input'>(
     )
 
     const handleClearButtonClick = useCallback(() => {
+      if (onChange) {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        onChange({ target: { value: '' } })
```

Comment:
if this is controlled, why is it always firing with `''` ah? just curious

Reply:
Since its controlled, we need to use `onChange` to clear the external state when the clear button is pressed so we send `''`. For all other purposes, the `onChange` function is just passed to the `Input` component which will handle everything else.

Reply:
Should we also use the `useControllableState` hook to convert all uncontrolled/controlled props passed into the component into a controlled component? Then we can handle the current value much better.